pkg/users/accounts-list.js

```diff
@@ -18,72 +18,277 @@
+    constructor(props) {
+        super(props);
+        this.state = {
+            currentTextFilter: '',
+        };
+
+        this.setCurrentTextFilter = this.setCurrentTextFilter.bind(this);
+    }
+
+    setCurrentTextFilter(currentTextFilter) {
+        this.setState({ currentTextFilter });
+    }
+
+    render() {
+        const accounts = mapGroupsToAccount(this.props.accounts, this.props.groups);
+
+        const filtered_accounts = accounts.filter(account => {
+            if ((account.uid < 1000 && account.uid !== 0) ||
+                     account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
+                     account.shell === '/bin/false')
+                return false;
+
+            const { currentTextFilter } = this.state;
+
+            if (currentTextFilter !== "" &&
+                (account.name.toLowerCase().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+                (account.gecos.toLowerCase().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+                (account.uid.toString().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+                (!account.groups.find(group => group.toLowerCase().indexOf(currentTextFilter.toLowerCase()) !== -1)))
+                return false;
+
+            return true;
+        });
 
-    return (
-        <Page id="accounts">
-            { superuser.allowed &&
-                <PageSection variant={PageSectionVariants.light}>
-                    <Button id="accounts-create" onClick={() => account_create_dialog(accounts)}>
-                        {_("Create new account")}
-                    </Button>
+        const columns = [
+            { title: _('Username'), sortable: true },
+            { title: _('Full Name'), sortable: true },
+            { title: _('ID'), sortable: true },
+            { title: _('Last Active'), sortable: true },
+            { title: _('Group') },
+        ];
+
+        const sortRows = (rows, direction, idx) => {
+            const sortedRows = rows.sort((a, b) => {
+                const aitem = a.columns[idx];
+                const bitem = b.columns[idx];
+                const aname = a.columns[0];
+                const bname = b.columns[0];
+
+                // current user is always first
+                if (aname.sortKey === this.props.current_user)
+                    return direction === SortByDirection.asc ? -1 : 1;
+                if (bname.sortKey === this.props.current_user)
+                    return direction === SortByDirection.asc ? 1 : -1;
+                // sorting last login
+                if (idx === 3) {
+                    if (aitem.sortKey === "logged in")
+                        return -1;
+                    if (bitem.sortKey === "logged in")
+                        return 1;
+                    if (aitem.sortKey === "never")
+                        return 1;
+                    if (bitem.sortKey === "never")
+                        return -1;
```

Comment:
These 9 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17830-20221116-150219-482f17ff-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)